datastore/datastore.go

```diff
@@ -185,11 +186,14 @@ func ParseKey(key string) ([]string, error) {
 }
 
 // newClient used to connect to KV Store
-func newClient(scope string, kv string, addr string, config *store.Config, cached bool) (DataStore, error) {
+func newClient(scope string, kv string, addr string, config *store.Config, cached bool, sequential bool) (DataStore, error) {
 
 	if cached && scope != LocalScope {
 		return nil, fmt.Errorf("caching supported only for scope %s", LocalScope)
 	}
+	if scope == LocalScope && !sequential {
```

Comment:
@jimmycmh I think we can always make the change if requirements change in future, being an internal decision to this package.
If you could make the change, we could get you fix in docker 1.12.
Thanks.